instrumentation/http/src/main/java/brave/http/HttpResponse.java

```diff
@@ -28,31 +28,45 @@
  */
 public abstract class HttpResponse extends Response {
   /**
-   * Like {@link HttpRequest#method()} except used in response parsing.
+   * The request that initiated this HTTP response or {@code null} if unknown.
    *
-   * <p>Notably, this is used to create a route-based span name.
+   * <p>Implementations should return the last wire-level request that caused this response or
+   * error. HTTP properties like {@linkplain HttpRequest#path() path} and {@linkplain
+   * HttpRequest#header(String) headers} might be different, due to redirects or authentication.
+   * Some properties might not be visible until response processing, notably {@link #route()}.
    *
    * @since 5.10
    */
-  // TODO: Consider `httpResponse.request()` and deprecating `httpResponse.method()` #1086
-  @Nullable public String method() {
+  @Nullable public HttpRequest request() {
```

Comment:
@anuraaga I feel like this should be on Response, not HttpResponse. We added Response in 5.10 (this pending version). If we pull it up afterwards (for Rpc) it will look awkward I think..

Reply:
Ah yeah that makes sense to me